### PR TITLE
TST: mark two optimize.linprog tests as xfail. See gh-7877.

### PR DESCRIPTION
**Pull request by rgommers**
Also fix up an assert to be more informative.

### CODE REVIEW

File: scipy/optimize/tests/test_linprog.py [hunk 1]
```diff
@@ -904,6 +910,11 @@ def f():
 class TestLinprogIPSparse(BaseTestLinprogIP):
     options = {"sparse": True}
 
+    @pytest.mark.xfail(reason='Fails with ATLAS, see gh-7877')
+    def test_bug_6690(self):
+        # Test defined in base class, but can't mark as xfail there
+        pass
```
Comment (pv) [inline scipy/optimize/tests/test_linprog.py]: Maybe add `super(TestLinprogIPSparse, self).test_bug_6690()` so we get the xpass status when it passes?

Comment (rgommers) [reply scipy/optimize/tests/test_linprog.py]: Good idea, done.

File: scipy/optimize/tests/test_linprog.py [hunk 2]
```diff
@@ -937,3 +948,8 @@ class TestLinprogIPDense(BaseTestLinprogIP):
 
 class TestLinprogIPSparsePresolve(BaseTestLinprogIP):
     options = {"sparse": True, "_sparse_presolve": True}
+
+    @pytest.mark.xfail(reason='Fails with ATLAS, see gh-7877')
+    def test_bug_6690(self):
+        # Test defined in base class, but can't mark as xfail there
+        super(TestLinprogIPSparse, self).test_bug_6690()
```
Comment (pv) [inline scipy/optimize/tests/test_linprog.py]: TestLinprogIPSparsePresolve

Comment (pv) [reply scipy/optimize/tests/test_linprog.py]: ok, fixed it myself

Comment (rgommers) [reply scipy/optimize/tests/test_linprog.py]: thanks

**Comment by rgommers** [comment]
All green, in it goes. Thank @pv

**Comment by rgommers** [comment]
Backport in gh-7938

**Pull request merged**